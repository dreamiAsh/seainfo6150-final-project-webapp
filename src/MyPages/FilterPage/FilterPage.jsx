import React, {useEffect, useState} from "react";
import styles from "./FilterPage.module.css";
import { Switch, Route, Link } from "react-router-dom";
import RecipeList from "../../Components/RecipeList/RecipeList"

const FilterPage = (props) => {
    const [clicked, setClicked] = useState();
    function onClick(e) {
        setClicked(true)
    }

    return (
        clicked ? (<div>
                        <div className={styles.box}>
                            <button className={styles.button} onClick={onClick}>Fast food</button>
                            <button className={styles.button} onClick={onClick}>Dessert</button>
                            <button className={styles.button} onClick={onClick}>Lamp</button>
                        </div>
                        <RecipeList/>
                    </div>)
          :
          (<div className={styles.box}>
                <button className={styles.button} onClick={onClick}>Fast food</button>
                <button className={styles.button} onClick={onClick}>Dessert</button>
                <button className={styles.button} onClick={onClick}>Lamp</button>
          </div>)
    );
};

export default FilterPage;
