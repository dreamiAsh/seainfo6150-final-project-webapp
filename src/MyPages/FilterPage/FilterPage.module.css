.box{
    margin: 100px 100px;
    width: 80%;
    display: grid;
    flex-flow: column-reverse;
    justify-content: space-around;
    align-items: stretch;
    grid-template-columns: repeat(2, 1fr);
}
.button{
    width: 100px;
    height: 44px;
    background:  #E5B687;
    border-radius: 13px;
    border: none;
    margin-bottom: 20px;
    font-weight: bold;
}
.clicked{
    width: 100px;
    height: 44px;
    background:  orange;
    border-radius: 13px;
    border: none;
    margin-bottom: 20px;
    font-weight: bold;
}
.button:hover{
    background: orange;
}

@media (min-width: 600px) {
    .box {
        grid-template-columns: repeat(4, 1fr);
        font-size: 14px;
    }
}
@media (min-width: 800px) {
    .box {
        grid-template-columns: repeat(6, 1fr);
        font-size: 12px;
    }
}
@media (min-width: 1100px) {
    .box {
        grid-template-columns: repeat(8, 1fr);
        font-size: 11px;
    }
}