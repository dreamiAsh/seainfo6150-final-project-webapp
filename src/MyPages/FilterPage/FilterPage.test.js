import React from "react";
import { render, fireEvent } from "@testing-library/react";
import FilterPage from "./FilterPage.jsx";

it("renders the FilterPage component correctly", () => {
  const { container } = render(<FilterPage />);
  expect(container).toMatchSnapshot();
});