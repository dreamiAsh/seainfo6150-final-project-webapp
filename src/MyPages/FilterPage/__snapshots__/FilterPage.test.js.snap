// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the FilterPage component correctly 1`] = `
<div>
  <div
    class="box"
  >
    <button
      class="button"
    >
      Fast food
    </button>
    <button
      class="button"
    >
      Dessert
    </button>
    <button
      class="button"
    >
      Lamp
    </button>
  </div>
</div>
`;
